src/mame/machine/nes.cpp

```diff
@@ -70,28 +70,73 @@ void nes_state::machine_start()
 		m_ppu->set_latch(*m_cartslot->m_cart, FUNC(device_nes_cart_interface::ppu_latch));
 
 		// install additional handlers (read_h, read_ex, write_ex)
-		if (m_cartslot->get_pcb_id() == STD_EXROM || m_cartslot->get_pcb_id() == STD_NROM368 || m_cartslot->get_pcb_id() == STD_DISKSYS
-			|| m_cartslot->get_pcb_id() == GG_NROM || m_cartslot->get_pcb_id() == CAMERICA_ALADDIN || m_cartslot->get_pcb_id() == SUNSOFT_DCS
-			|| m_cartslot->get_pcb_id() == BANDAI_DATACH || m_cartslot->get_pcb_id() == BANDAI_KARAOKE || m_cartslot->get_pcb_id() == UNL_2A03PURITANS || m_cartslot->get_pcb_id() == AVE_MAXI15 || m_cartslot->get_pcb_id() == BTL_PALTHENA
-			|| m_cartslot->get_pcb_id() == KAISER_KS7010 || m_cartslot->get_pcb_id() == KAISER_KS7022 || m_cartslot->get_pcb_id() == KAISER_KS7030 || m_cartslot->get_pcb_id() == KAISER_KS7031 || m_cartslot->get_pcb_id() == KAISER_KS7037 || m_cartslot->get_pcb_id() == KAISER_KS7057 || m_cartslot->get_pcb_id() == BMC_VT5201
-			|| m_cartslot->get_pcb_id() == UNL_LH32 || m_cartslot->get_pcb_id() == UNL_LH10 || m_cartslot->get_pcb_id() == UNL_2708 || m_cartslot->get_pcb_id() == UNL_RT01
-			|| m_cartslot->get_pcb_id() == UNL_43272 || m_cartslot->get_pcb_id() == UNL_EH8813A || m_cartslot->get_pcb_id() == BMC_G63IN1 || m_cartslot->get_pcb_id() == BMC_8157
-			|| m_cartslot->get_pcb_id() == BMC_GOLD150 || m_cartslot->get_pcb_id() == BMC_CH001
-			|| m_cartslot->get_pcb_id() == BMC_70IN1 || m_cartslot->get_pcb_id() == BMC_800IN1)
+		static const int r_h_pcbs[] =
+		{
+			AVE_MAXI15,
+			BANDAI_DATACH,
+			BANDAI_KARAOKE,
+			BMC_70IN1,
+			BMC_800IN1,
+			BMC_8157,
+			BMC_CH001,
+			BMC_G63IN1,
+			BMC_GOLD150,
+			BMC_VT5201,
+			BTL_PALTHENA,
+			CAMERICA_ALADDIN,
+			GG_NROM,
+			KAISER_KS7010,
+			KAISER_KS7022,
+			KAISER_KS7030,
+			KAISER_KS7031,
+			KAISER_KS7037,
+			KAISER_KS7057,
+			SACHEN_3013,
+			STD_DISKSYS,
+			STD_EXROM,
+			STD_NROM368,
+			SUNSOFT_DCS,
+			UNL_2708,
+			UNL_2A03PURITANS,
+			UNL_43272,
+			UNL_EH8813A,
+			UNL_LH10,
+			UNL_LH32,
+			UNL_RT01
+		};
+
+		static const int w_ex_pcbs[] =
+		{
+			BTL_09034A,
+			BTL_SMB2JB,
+			UNL_AC08,
+			UNL_SMB2J
+		};
+
+		static const int rw_ex_pcbs[] =
+		{
+			KAISER_KS7017,
+			STD_DISKSYS,
+			UNL_603_5052
+		};
+
+		int pcb_id = m_cartslot->get_pcb_id();
+
+		if (std::find(std::begin(r_h_pcbs), std::end(r_h_pcbs), pcb_id) != std::end(r_h_pcbs))
```

Comment:
The approach of having the handlers installed here isn’t viable long-term.  It’s going to change for the better.  I’m probably going to have to do it in a way that allows devices to be converted piecemeal, there are just so many NES cartridge types.

Reply:
I'm not surprised to hear it. Still in the interim would this change be ok? It's functionally identical and done for ease of insertion and readability (including PR diffs!). I just got sick of looking at the verbose unordered wall of ORs.

BTW, it's hard to tell but SACHEN_3013 is the new entry here. It's the fifth one that's needed a bespoke read_h since I started adding NES boards about a month ago, and I suspect there are many more ad-hoc Asian bootleg cartridges to come in this category.